File: csharp/collatz-conjecture/CollatzConjectureTests.cs
using System;
using Xunit;

public class CollatzConjectureTests
{
    [Fact]
    public void Zero_steps_for_one()
    {
        Assert.Equal(0, CollatzConjecture.Steps(1));
    }

    [Fact]
    public void Divide_if_even()
    {
        Assert.Equal(4, CollatzConjecture.Steps(16));
    }

    [Fact]
    public void Even_and_odd_steps()
    {
        Assert.Equal(9, CollatzConjecture.Steps(12));
    }

    [Fact]
    public void Large_number_of_even_and_odd_steps()
    {
        Assert.Equal(152, CollatzConjecture.Steps(1000000));
    }

    [Fact]
    public void Zero_is_an_error()
    {
        Assert.Throws<ArgumentOutOfRangeException>(() => CollatzConjecture.Steps(0));
    }

    [Fact]
    public void Negative_value_is_an_error()
    {
        Assert.Throws<ArgumentOutOfRangeException>(() => CollatzConjecture.Steps(-15));
    }
}

File: vbnet/collatz-conjecture/CollatzConjectureTests.vb
Imports System
Imports Xunit
Public Class CollatzConjectureTests
    <Fact>
    Public Sub ZeroStepsForOne()
        Assert.Equal(0, Steps(1))
    End Sub
    <Fact>
    Public Sub DivideIfEven()
        Assert.Equal(4, Steps(16))
    End Sub
    <Fact>
    Public Sub EvenAndOddSteps()
        Assert.Equal(9, Steps(12))
    End Sub
    <Fact>
    Public Sub LargeNumberOfEvenAndOddSteps()
        Assert.Equal(152, Steps(1000000))
    End Sub
    <Fact>
    Public Sub ZeroIsAnError()
        Assert.Throws(Of ArgumentOutOfRangeException)(Function() Steps(0))
    End Sub
    <Fact>
    Public Sub NegativeValueIsAnError()
        Assert.Throws(Of ArgumentOutOfRangeException)(Function() Steps(-15))
    End Sub
End Class